core/lib/src/local/blocking/client.rs

```diff
@@ -0,0 +1,316 @@
+use std::borrow::Cow;
+
+use tokio::sync::oneshot::{channel, Sender};
+
+use crate::error::LaunchError;
+use crate::http::Method;
+use crate::local::blocking::LocalRequest;
+use crate::rocket::{Rocket, Manifest};
+
+pub struct Client {
+    pub(crate) inner: crate::local::Client,
+    // Handle to the Runtime used for executing tasks
+    handle: tokio::runtime::Handle,
+    // Runtime shutdown signal. When this sender is dropped, the Receive half
+    // will Error and shut down the runtime.
+    _shutdown_tx: Sender<std::convert::Infallible>,
+}
+
+impl Client {
+    fn _new(rocket: Rocket, tracked: bool) -> Result<Client, LaunchError> {
+        let mut runtime = tokio::runtime::Builder::new()
+            .basic_scheduler()
+            .enable_all()
+            .build()
+            .expect("create tokio runtime");
+
+        let (_shutdown_tx, shutdown_rx) = channel();
+        let handle = runtime.handle().clone();
+
+        // Initialize the Rocket instance
+        let inner = runtime.block_on(crate::local::Client::_new(rocket, tracked))?;
+
+        // Spawn a background thread to drive the single-threaded runtime
+        std::thread::spawn(move || {
```

Comment:
Why is this necessary? See my top-level comment.

Reply:
Ah, I had this case in the back of my mind but did not actually write any example using it: suppose I have an `Arc<blocking::Client>`, and I want to run one request each in two separate threads? With the background thread doing the actual work, those requests will complete "as concurrently as possible". With the `Mutex`, the same requests will complete one at a time.

Since there's no I/O with the client, the only way way this makes a difference is for routes that use the database or timers... after reconsidering I would rather adopt the `Mutex` - "use the non-blocking `Client` if you want concurrency" seems a fair tradeoff. And I imagine the `Mutex` should be faster for the (more common) single-thread usage.